lib/BackgroundJob/Tasks/CreateClustersTask.php

```diff
@@ -290,20 +295,23 @@ private function getNewClusters(array $faces): array {
 				}
 			}
 		} else {
-			for ($i = 0, $face_count1 = count($faces); $i < $face_count1; $i++) {
+			$euclidean = new Euclidean();
+			$faces_count = count($faces);
+			for ($i = 0; $i < $faces_count; $i++) {
 				$face1 = $faces[$i];
-				if ($face1->confidence < $min_confidence) {
+				if (($face1->confidence < $min_confidence) ||
+				    ($face1->height() < $min_face_size)) {
```

Comment:
why do you compare height only? Is our model always returning height>width for found faces? In theory, you probably want here `max($face1->height(), $face1->width()) < $min_face_size`? What about orientation, if image is landscape, is width>height for found faces, or it is already accounted (rotated)?

Reply:
Just to optimize (Although probably exaggerated ..), and because I assume that the faces are always vertical perhaps change it.
But in general the rectangles are quite square, and if not, it is another poor quality alarm. :thinking:

Reply:
> But in general the rectangles are quite square, and if not, it is another poor quality alarm. thinking

Check my database, and the only faces that make a big difference are that they are either clipped on the edge, or they are just bad and have low confidence.

Reply:
Done..